numpy/distutils/fcompiler/intel.py

```diff
@@ -15,6 +15,17 @@ def intel_version_match(type):
 
 
 class BaseIntelFCompiler(FCompiler):
+    def __init__(self, *args, **kwargs):
+        FCompiler.__init__(self, *args, **kwargs)
+        # specify version discovery bits
+        if platform.system() == "Windows":
+            self.executables['version_cmd'] = ["icl" , "-dummy"]
+            self.version_pattern = r'Version\s*(?"version"[\d.]+)'
+        else:
+            self.executables['version_cmd'] = ["icc", "-dumpversion"]
+            self.version_pattern = r'^(?"version"[\d.]$)'
```

Comment:
Similarly, the subclasses already define `version_match`, which takes precedence over `version_pattern`.